### ISSUE TITLE
Fails with latest Snap! version

### ISSUE THREAD

**Issue by bromagosa** [Rydion/SnapSubset#1]
The latest Snap! version introduces modifications that SnapSubset can't seem to handle right now. I've tried to modify a couple of _info_ files myself, but I haven't managed to get it to work...

Here's the output:

```
java.nio.charset.MalformedInputException: Input length = 2
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 2
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
line 1:0 mismatched input '*' expecting {<EOF>, PRAGMA, STRING, REGULAR_EXPRESSION, 'new', 'true', 'false', 'var', '[', '{', '(', 'function', NUMBER, ID}
java.nio.charset.MalformedInputException: Input length = 2
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.nio.charset.MalformedInputException: Input length = 2
    at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.io.BufferedReader.readLine(BufferedReader.java:324)
    at java.io.BufferedReader.readLine(BufferedReader.java:389)
    at java.nio.file.Files.readAllLines(Files.java:3205)
    at main.SnapSubset.lambda$0(SnapSubset.java:229)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Iterator.forEachRemaining(Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    at main.SnapSubset.main(SnapSubset.java:224)
java.lang.NullPointerException
    at dataStructures.ClassInfo.<init>(ClassInfo.java:53)
    at parser.listeners.WalkListener.getParameterFromLeftObjectContext(WalkListener.java:177)
    at parser.listeners.WalkListener.enterFunction_declaration_util(WalkListener.java:1011)
    at antlr4.SnapGrammarParser$Function_declaration_utilContext.enterRule(SnapGrammarParser.java:1515)
    at parser.walkers.SnapTreeWalker.enterRule(SnapTreeWalker.java:89)
    at parser.walkers.SnapTreeWalker.walk(SnapTreeWalker.java:57)
    at parser.listeners.WalkListener.walkRule(WalkListener.java:148)
    at parser.listeners.WalkListener.enterAssign_util(WalkListener.java:570)
    at antlr4.SnapGrammarParser$Assign_utilContext.enterRule(SnapGrammarParser.java:932)
    at parser.walkers.SnapTreeWalker.enterRule(SnapTreeWalker.java:89)
    at parser.walkers.SnapTreeWalker.walk(SnapTreeWalker.java:57)
    at parser.walkers.SnapTreeWalker.walk(SnapTreeWalker.java:70)
    at main.SnapSubset.analyze(SnapSubset.java:109)
    at main.SnapSubset.main(SnapSubset.java:271)
FAULTY PARAMETER = forClone
```

So I thought I understood there was a missing _forClone_ parameter somewhere, and since this parameter is only present in _blocks.js_, I modified _blocks.info_ accordingly to reflect the changes:

```
{name : fullCopy, class : Function, return : BlockMorph, parameters: {
    {forClone : Boolean}
}},
```

But I still get the same error, so I guess I don't really know how _info_ files actually work.

**Comment by adrian-hintze** [comment]
Hmm, can't tell for sure what's happening. I'll take a look at it these weekend.

**Comment by adrian-hintze** [comment]
Hi, so you were indeed on the right track @bromagosa, but the syntax wasn't quite right, the correct addition would be:

```
{name : fullCopy, class : Function, return : BlockMorph, parameters: {
    {name: forClone, class: Boolean}
}},
```

Anyways, I downloaded the latest Snap! version and there have been A LOT of changes since I last generated Snapp's reduced execution environment. I'll update all the info files and commit them for testing, but I'll need a few days to get it all done.

**Comment by bromagosa** [comment]
Thanks, @Rydion!

**Comment by adrian-hintze** [comment]
Hi @bromagosa, I've been way busier than expected lately, but over the course of this next week the update should be finished. It would be awesome if you could provide me with some test/sample projects that make use of the new Snap! features (like tables), that way I can make sure that I don't miss any important functionality. Thank you.

**Comment by bromagosa** [comment]
Sure! I'll build some new test projects. Actually, @jmoenig has examples for all new features, so I'll search among those! :)

**Comment by adrian-hintze** [comment]
Hi!
I've updated the info files to work with the latest Snap! Master branch and generated a new Snap! subset (can be found in the new folder "working - beta releases"). I've tested it with all the projects that I have available and it seems to work just fine, but, as I said, Snap!'s source code has changed a lot and I had to finetune some of the info files quite a bit. I'm not 100% confident that it will work perfectly right of the bat so I would like to test it a bit more before pushing it to Snapp!.

If over the course of next week we can test this thoroughly (using new, relevant test projects) and everything works ok, I'll update the http://snapp.citilab.eu/ version of Snapp! with the new files and also push some bug fixes while I'm at it.

**Comment by bromagosa** [comment]
Yep! It seems to work fine! I'm trying different projects now, but I believe it could be pushed to the Snapp! site already!

**Comment by bromagosa** [comment]
So, @jmoenig's collection can be found here: http://snap.berkeley.edu/SnapExamples.pdf

The first example that uses new features (Path Follower) doesn't work with the 29_05_2016 beta release. I don't understand why, because the ``reportGet`` function is present in ``threads.js`` and its corresponding block spec is properly defined in ``objects.js``.

However, the ones I thought wouldn't work (like Population Simulation) are running fine.

**Comment by bromagosa** [comment]
So, @jmoenig's collection can be found here: http://snap.berkeley.edu/SnapExamples.pdf

The first example that uses new features ([Path Follower](http://snap.berkeley.edu/snapsource/snap.html#cloud:Username=jens&ProjectName=PathFollower)) doesn't work with the 29_05_2016 beta release. I don't understand why, because the reportGet function is present in threads.js and its corresponding block spec is properly defined in objects.js.

[Population Simulation](http://snap.berkeley.edu/snapsource/snap.html#cloud:Username=jens&ProjectName=population) doesn't work either, and I guess it may also the case for all projects using features related to first class sprites.

**Comment by adrian-hintze** [comment]
Hi, thank you for the feedback! I'll try to sort these problems out and if those are the only issues I'll push the update to Snapp! once I get it to work.

**Comment by Rydion** [comment]
By the way, this hasn't anything to do with this issue, but while we're at it... Snapp! keeps

**Comment by adrian-hintze** [comment]
Hi. So, I sorted out the problems with first class sprites. I tested everything and it seems to be working just fine. As the update should be stable enough now I already pushed it to http://snapp.citilab.eu/, I have tested it with all the sample projects and everything works ok, but as always, maybe there are some edge cases where it doesn't, so a bit more testing wouldn't hurt anybody.

Also, I found a problem with NW.js and mp3 sound files, but that would be an issue for [Snapp!](https://github.com/Rydion/Snapp), as SnapSubset doesn't rely on it.

P.S.: Every once in a while Snapp! stops working and I need to manually go to the server and run my startup script. I'm guessing this happens when the server restarts. Would it be possible to automate this process?
